borg/_hashindex.c

```diff
@@ -400,6 +435,15 @@ hashindex_set(HashIndex *index, const void *key, const void *value)
         while(!BUCKET_IS_EMPTY(index, idx) && !BUCKET_IS_DELETED(index, idx)) {
             idx = (idx + 1) % index->num_buckets;
         }
+        if(BUCKET_IS_EMPTY(index, idx)){
+            index->num_empty--;
```

Comment:
don't you want to also `index->num_empty++` in case you're inserting on top of a tombstone?